lib/charms/mongodb_libs/v0/mongodb_provider.py

```diff
@@ -192,7 +195,7 @@ def _get_database_from_relation(self, relation: Relation) -> Optional[str]:
             if unit.app is self.charm.app:
                 # it is peer relation, skip
                 continue
-            database = relation.data[unit].get("database", None)
+            database = relation.data[relation.app].get("database", None)
```

Comment:
in such a case, we don't need to iterate via `relation.units` can you please remove lines 191-194

Reply:
Gotcha! Great catch, thanks @delgod